rero_ils/modules/locations/permissions.py

```diff
@@ -15,47 +16,89 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-"""Location permissions."""
+"""Permissions for locations."""
 
+from rero_ils.modules.organisations.api import current_organisation
+from rero_ils.modules.patrons.api import current_patron
+from rero_ils.modules.permissions import RecordPermission
 
-from ...permissions import staffer_is_authenticated
 
+class LocationPermission(RecordPermission):
+    """Location permissions."""
 
-def can_update_delete_location_factory(record, *args, **kwargs):
-    """Checks if logged user can update or delete its organisation locations.
+    @classmethod
+    def list(cls, user, record=None):
+        """List permission check.
 
-    user must have librarian or system_librarian role
-    librarian can only update or delete its affiliated library locations.
-    sys_librarian can update or delete any location of its organisation.
-    """
-    def can(self):
-        patron = staffer_is_authenticated()
-        if patron and patron.organisation_pid == record.organisation_pid:
-            if not patron.is_system_librarian:
-                if patron.library_pid and \
-                        record.library_pid != patron.library_pid:
-                    return False
-            return True
-        return False
-    return type('Check', (), {'can': can})()
+        :param user: Logged user.
+        :param record: Record to check.
+        :return: True is action can be done.
+        """
+        # List locations allowed only for staff members (lib, sys_lib)
+        return current_patron and current_patron.is_librarian
 
+    @classmethod
+    def read(cls, user, record):
+        """Read permission check.
 
-def can_create_location_factory(record, *args, **kwargs):
-    """Checks if the logged user can create locations of its organisation.
+        :param user: Logged user.
+        :param record: Record to check.
+        :return: True is action can be done.
+        """
+        if not current_patron:
+            return False
+        # only staff members (lib, sys_lib) are allowed to read an location
+        if not current_patron.is_librarian:
+            return False
```

Comment:
Why not do things as l.65? `if not current_patron or not current_patron.is_librarian:`

Reply:
done